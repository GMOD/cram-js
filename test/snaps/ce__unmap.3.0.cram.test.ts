import { testFile } from '../testUtil'
testFile('ce#unmap.3.0.cram')
