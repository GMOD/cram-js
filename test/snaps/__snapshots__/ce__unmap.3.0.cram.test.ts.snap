// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole ce#unmap.3.0.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 44,
      "_size": 18,
      "alignmentSpan": 0,
      "crc32": 1304357715,
      "landmarks": [
        0,
        33,
      ],
      "length": 58,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 44,
        "_size": 18,
        "alignmentSpan": 0,
        "crc32": 1304357715,
        "landmarks": [
          0,
          33,
        ],
        "length": 58,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [],
      },
      {
        "samHeader": [],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BA": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 30,
          },
          "parametersBytes": 1,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              4,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              100,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 125,
      "_size": 23,
      "alignmentSpan": 1,
      "crc32": 2460692403,
      "landmarks": [
        181,
      ],
      "length": 438,
      "numBases": 600,
      "numBlocks": 6,
      "numLandmarks": 1,
      "numRecords": 6,
      "recordCounter": 0,
      "refSeqId": -1,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 125,
        "_size": 23,
        "alignmentSpan": 1,
        "crc32": 2460692403,
        "landmarks": [
          181,
        ],
        "length": 438,
        "numBases": 600,
        "numBlocks": 6,
        "numLandmarks": 1,
        "numRecords": 6,
        "recordCounter": 0,
        "refSeqId": -1,
        "refSeqStart": 0,
      },
      {
        "features": [
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": "CCTAGCCCTAACCCTAACCCTAACCCTAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAA",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "I",
            "sequenceId": -1,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 312,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": "CCTAGCCCTAACCCTAACCCTAACCCTAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAA",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "II",
            "sequenceId": -1,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 313,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": "CCTAGCCCTAACCCTAACCCTAACCCTAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAA",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "III",
            "sequenceId": -1,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 314,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": "CCTAGCCCTAACCCTAACCCTAACCCTAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAA",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "IV",
            "sequenceId": -1,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 315,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": "CCTAGCCCTAACCCTAACCCTAACCCTAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAAGCCTAA",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "V",
            "sequenceId": -1,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 316,
          },
          {
            "alignmentStart": 0,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": "ACTAAGCCTAAGCCTAAGCCTAAGCCAATTATCGATTTCTGAAAAAATTATCGAATTTTCTAGAAATTTTGCAAATTTTTTCATAAAATTATCGATTTTA",
            "readGroupId": -1,
            "readLength": 100,
            "readName": "VI",
            "sequenceId": -1,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 317,
          },
        ],
        "header": {
          "_endPosition": 350,
          "_size": 39,
          "compressedSize": 35,
          "compressionMethod": "raw",
          "content": Uint8Array [
            255,
            255,
            255,
            255,
            15,
            0,
            1,
            6,
            0,
            4,
            3,
            11,
            12,
            30,
            255,
            255,
            255,
            255,
            15,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
          ],
          "contentId": 0,
          "contentPosition": 311,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 3614694957,
          "parsedContent": {
            "_endPosition": 160,
            "_size": 35,
            "contentIds": [
              11,
              12,
              30,
            ],
            "md5": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "numBlocks": 4,
            "numContentIds": 3,
            "numRecords": 6,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": -1,
            "refSeqSpan": 1,
            "refSeqStart": 0,
          },
          "uncompressedSize": 35,
        },
      },
    ],
  },
]
`;
